src/openMVG/sfm/pipelines/sequential/sequential_SfM.cpp

```diff
@@ -110,42 +194,70 @@ void SequentialSfMReconstructionEngine::RobustResectionOfImages(
-
-      // std::cout << "Global Bundle start, resection group index: " << resectionGroupIndex << ".\n";
+      if((resectionGroupIndex % 10) == 0)
+      {
+        chrono_start = std::chrono::steady_clock::now();
+        // Scene logging as ply for visual debug
+        std::ostringstream os;
+        os << std::setw(8) << std::setfill('0') << resectionGroupIndex << "_Resection";
+        Save(_sfm_data, stlplus::create_filespec(_sOutDirectory, os.str(), _sfmdataInterFileExtension), _sfmdataInterFilter);
+        std::cout << "Save of file " << os.str() << " took " << std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::steady_clock::now() - chrono_start).count() << " msec." << std::endl;
+      }
+      std::cout << "Global Bundle start, resection group index: " << resectionGroupIndex << ".\n";
+      chrono_start = std::chrono::steady_clock::now();
       int bundleAdjustmentIteration = 0;
+      const std::size_t nbOutliersThreshold = 50;
       // Perform BA until all point are under the given precision
       do
       {
-        // std::cout << "Resection group index: " << resectionGroupIndex << ", bundle iteration: " << bundleAdjustmentIteration << "\n";
+        auto chrono2_start = std::chrono::steady_clock::now();
         BundleAdjustment();
+        std::cout << "Resection group index: " << resectionGroupIndex << ", bundle iteration: " << bundleAdjustmentIteration
+                  << " took " << std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::steady_clock::now() - chrono2_start).count() << " msec." << std::endl;
         ++bundleAdjustmentIteration;
       }
-      while (badTrackRejector(4.0, 50) != 0);
+      while (badTrackRejector(4.0, nbOutliersThreshold) != 0);
+      std::cout << "Bundle with " << bundleAdjustmentIteration << " iterations took " << std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::steady_clock::now() - chrono_start).count() << " msec." << std::endl;
+      chrono_start = std::chrono::steady_clock::now();
+      eraseUnstablePosesAndObservations(this->_sfm_data, _minPointsPerPose, _minTrackLength);
+      std::cout << "eraseUnstablePosesAndObservations took " << std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::steady_clock::now() - chrono_start).count() << " msec." << std::endl;
     }
     ++resectionGroupIndex;
   }
   // Ensure there is no remaining outliers
   badTrackRejector(4.0, 0);
+  eraseUnstablePosesAndObservations(this->_sfm_data, _minPointsPerPose, _minTrackLength);
 }
 
 bool SequentialSfMReconstructionEngine::Process()
 {
+  // Update cache values
+  if(_pyramidWeights.size() != _pyramidDepth)
+  {
+    _pyramidWeights.resize(_pyramidDepth);
+    std::size_t maxWeight = 0;
+    for(std::size_t level = 0; level < _pyramidDepth; ++level)
+    {
+      std::size_t nbCells = Square(std::pow(2.0, level+1));
+      _pyramidWeights[level] = std::pow(2.0, (_pyramidDepth-(level+1)));
```

Comment:
better put a note that we are using  a different weighting trying to give more importance to the first levels of the pyramid.
The paper uses `w = 2^l`  (even if there is a typo in the text where they say to use w=2^{2*l}